source/remoteClient/connectionInfo.py

```diff
@@ -0,0 +1,97 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2015-2025 NV Access Limited, Christopher Toth, Tyler Spivey, Babbage B.V., David Sexton and others.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+from dataclasses import dataclass
+from enum import Enum
+from urllib.parse import parse_qs, urlencode, urlparse, urlunparse
+
+from . import protocol
+from .protocol import SERVER_PORT, URL_PREFIX
+
+
+class URLParsingError(Exception):
+	"""Raised if it's impossible to parse out the URL"""
+
+
+class ConnectionMode(Enum):
+	MASTER = "master"
+	SLAVE = "slave"
+
+
+class ConnectionState(Enum):
+	CONNECTED = "connected"
+	DISCONNECTED = "disconnected"
+	CONNECTING = "connecting"
+	DISCONNECTING = "disconnecting"
+
+
+@dataclass
+class ConnectionInfo:
+	hostname: str
+	mode: ConnectionMode
+	key: str
+	port: int = SERVER_PORT
+	insecure: bool = False
+
+	def __post_init__(self):
+		self.port = self.port or SERVER_PORT
+		self.mode = ConnectionMode(self.mode)
```

Comment:
This also seems a bit ambiguous to me. If people just pass a ConnectionType instance to the constructor, it should be OK, isn't it?